web/viewer.js

```diff
@@ -1615,11 +1620,19 @@ function webViewerInitialized() {
   document.getElementById('sidebarToggle').addEventListener('click',
     function() {
       this.classList.toggle('toggled');
+      var wasSidebarOpen = outerContainer.classList.contains('sidebarOpen');
+      PDFViewerApplication.sidebarOpen = !wasSidebarOpen;
+      if (PDFViewerApplication.sidebarOpen) {
+        sidebarContainer.classList.remove('hidden');
+      }
+      // The sidebar opening transition does not work if the 'hidden' class is
+      // still active for the sidebar container. The forceRendering() call
+      // between remove('hidden') and toogle('sidebarOpen') seems to provide
+      // enough time for the remove('hidden') statement to complete before
+      // opening the sidebar.
+      PDFViewerApplication.forceRendering();
       outerContainer.classList.add('sidebarMoving');
       outerContainer.classList.toggle('sidebarOpen');
```

Comment:
> Hmm, I played around with this patch and it happend one - and only one - time that the opening transition didn't work when opening the thumbnail bar ~50 times.

Would it also make sense to enclose the two lines above in a `setTimeout` to move them to the next turn of the event loop, to perhaps further reducing the risk of the open transition not working? I.e. like this
```
setTimeout(function () {
  outerContainer.classList.add('sidebarMoving');
  outerContainer.classList.toggle('sidebarOpen');
}, 0);